src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -277,14 +301,58 @@ object DataTypes {
       predicate.map(_.into[Predicate].withFieldConst(_.field, field).transform)
   }
 
+  /** Class representing snapshot */
+  case class Snapshot(field: String, value: Any) {
+
+    /** Transforms snapshot to predicate */
+    def toPredicate: Predicate =
+      Predicate(
+        field = field,
+        operation = OperationType.after,
+        set = List(value),
+        inverse = true
+      )
+
+    /** Transforms snapshot to ordering */
+    def toOrdering: QueryOrdering =
+      QueryOrdering(
+        field = field,
+        direction = OrderDirection.desc
+      )
+  }
+
+  /** Finds invalid snapshot fields*/
+  def findInvalidSnapshotFields(
+      query: Query,
+      entity: EntityPath,
+      metadataConfiguration: MetadataConfiguration
+  ): List[InvalidSnapshotField] =
+    query.snapshot match {
```

Comment:
If you like to consider an alternative, you could try importing 
```scala
cats.instances.option._
cats.instances.list._

``` 

to have an extension `query.snapshot.foldMap` operation that would only need filling the mapping part, as the `List.empty` will be given for granted as the empty value for the list monoid.

Reply:
done